daras_ai_v2/base.py

```diff
@@ -569,7 +569,22 @@ def _render_share_modal(self, dialog: gui.AlertDialogRef):
                 checked_by_default=False,
             )
 
-        with gui.div(className="d-flex justify-content-between"):
+        workspaces = self.request.user.cached_workspaces
+        if workspaces and self.current_pr.workspace not in workspaces:
+            with gui.div(className="alert alert-warning mb-0 mt-4"):
+                duplicate = gui.button(
+                    f"{icons.fork} Duplicate", type="link", className="d-inline m-0 p-0"
+                )
+                gui.html(" this workflow to edit with others")
+                ref = gui.use_alert_dialog(key="publish-modal")
+                if duplicate:
+                    self.clear_publish_form()
+                    ref.set_open(True)
+                if ref.is_open:
+                    gui.session_state["published_run_workspace"] = workspaces[-1].id
```

Comment:
@devxpy having this assignment here creates the bug that no other workspace can be selected, because we overwrite it and `ref.is_open` is run every time after `ref.set_open(True)`.

we only want this to be the default - hence I had put this in the `if duplicate: ...` branch